app/src/main/java/mozilla/lockbox/store/AutoLockStore.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.store
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.SystemClock
+import android.preference.PreferenceManager
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.subjects.ReplaySubject
+import io.reactivex.subjects.Subject
+import mozilla.lockbox.action.LifecycleAction
+import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.support.Constant
+
+class AutoLockStore(
+    val dispatcher: Dispatcher = Dispatcher.shared,
+    val settingStore: SettingStore = SettingStore.shared,
+    val lifecycleStore: LifecycleStore = LifecycleStore.shared
+) : ContextStore {
+
+    companion object {
+        val shared by lazy { AutoLockStore() }
+    }
+
+    private val compositeDisposable = CompositeDisposable()
+    private lateinit var preferences: SharedPreferences
+
+    val lockRequired: Observable<Boolean> = ReplaySubject.createWithSize(1)
+
+    init {
+        lifecycleStore.lifecycleFilter
+            .filter { it == LifecycleAction.Foreground }
+            .subscribe {
+                checkElapsedTime()
+            }
+            .addTo(compositeDisposable)
+
+        lifecycleStore.lifecycleFilter
+            .filter { it == LifecycleAction.Background }
+            .switchMap { settingStore.autoLockTime.take(1) }
+            .subscribe {
+                val currentSystemTime = SystemClock.elapsedRealtime()
+                preferences
+                    .edit()
+                    .putLong(Constant.Key.autoLockTimerDate, currentSystemTime + it.ms)
+                    .apply()
+            }
```

Comment:
When to store the `boot_id` to compare here may end up being problematic, though I'd guess it were here.